src/main/java/org/coffeetrain/doggifs/GifGridView.kt

```diff
@@ -0,0 +1,14 @@
+package org.coffeetrain.doggifs
+
+import android.content.Context
+import android.graphics.BitmapFactory
+import android.widget.ImageView
+
+class GifGridView(context: Context, fileName: String) : ImageView(context) {
+  init {
+    scaleType = ScaleType.CENTER_CROP
+    val gifBytes = DogGifsApplication.getDogGifRepository(context).getGif(fileName)
```

Comment:
with the `operator get` on `DogGifRepository`, this could even be `context.dogGifs[filename]`.